docs/src/filters.md

```diff
@@ -0,0 +1,112 @@
+# Filters
+
+HDF5 supports filters for compression and validation: these are applied sequentially to
+each chunk of a dataset when writing data, and in reverse order when reading data.
+
+```@meta
+CurrentModule = HDF5
+```
+
+These can be set by passing a filter or vector of filters as a `filters` property to
+[`DatasetCreateProperties`](@ref).
+
+## Filter Interface
+
+```@meta
+CurrentModule = HDF5.Filters
+```
+
+The filter interface is used to describe filters and obtain information on them.
+
+```@docs
+Filters
+Filter
+FilterPipeline
+UnknownFilter
+FILTERS
+filterid
+encoder_present
+decoder_present
+filtername
+can_apply_func
+can_apply_cfunc
+set_local_func
+set_local_cfunc
+filter_func
+filter_cfunc
+register_filter
+```
```

Comment:
This is mostly internals: while good to document, it should probably be moved to the bottom.

Reply:
Moved in [4fff354](https://github.com/JuliaIO/HDF5.jl/pull/901/commits/4fff354772ecc80a1d51e0c32558a86f409f07c9)